app/src/main/java/org/coepi/android/cen/Utils.kt

```diff
@@ -0,0 +1,103 @@
+package org.coepi.android.cen
+
+import java.nio.ByteBuffer
+import java.nio.ByteOrder
+import java.security.MessageDigest
+import java.util.*
+
+@ExperimentalUnsignedTypes
+fun compareByteArray(a : UByteArray, b : UByteArray) : Int {
+    for (i in 0..a.size) {
```

Comment:
This is range is inclusive and would crash on the last `i`.
You can use `for (i in a.indices) { ... }` instead